### PR TITLE
Select the DeepSpeedCPUOptimizer based on the original optimizer class.

### PR DESCRIPTION
**Pull request by eljandoubi**
# What does this PR do?

For DeepSpeed optimizer offloading, select the appropriate optimizer based on the original optimizer class.

## Who can review?

- DeepSpeed: @muellerzr
- Core parts of the library: @muellerzr @BenjaminBossan @SunMarc

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
Thanks for this update. Just wondering aloud: If the optimizer is neither Adam, nor Adagrad, nor Lion, what would be the right thing to do? As is, Adam is used, is that correct?

**Comment by eljandoubi** [comment]
@BenjaminBossan By défaut, I have kept the Adam as it was for everyone torch optimizer.

File: src/accelerate/accelerator.py [hunk 1]
```diff
@@ -1839,10 +1841,42 @@ def _prepare_deepspeed(self, *args):
...
+                        # For DeepSpeedCPULion
+                        if is_bnb_available(min_version="0.38.0") and compare_versions("deepspeed", ">=", "0.11.0"):
+                            from bitsandbytes.optim import Lion, Lion32bit
+
+                            if isinstance(optimizer, (Lion, Lion32bit)) and optimizer.optim_bits == 32:
+                                from deepspeed.ops.lion import DeepSpeedCPULion
+
+                                optimizer_class = DeepSpeedCPULion
+
+                        optimizer = optimizer_class(optimizer.param_groups, **defaults)
```
Comment (muellerzr) [inline src/accelerate/accelerator.py:1879]: This seems a bit bulky for the `Accelerator`, can we abstract this out to a deepspeed util of `map_pytorch_optim_to_deepspeed`?

**Review by muellerzr — COMMENTED** [review]
Thanks for doing this! Overall I'm a fan, however let's abstract this out to a util func to keep the Accelerator a bit more readable

**Comment by eljandoubi** [comment]
@muellerzr What do you think of it like this?

**Pull request merged**